src/protocol/channels.ts

```diff
@@ -584,12 +584,13 @@ export type BrowserContextInitializer = {
 };
 export interface BrowserContextChannel extends Channel {
   on(event: 'bindingCall', callback: (params: BrowserContextBindingCallEvent) => void): this;
-  on(event: 'close', callback: (params: BrowserContextCloseEvent) => void): this;
+  on(event: 'close', scallback: (params: BrowserContextCloseEvent) => void): this;
   on(event: 'page', callback: (params: BrowserContextPageEvent) => void): this;
   on(event: 'route', callback: (params: BrowserContextRouteEvent) => void): this;
   on(event: 'video', callback: (params: BrowserContextVideoEvent) => void): this;
   on(event: 'backgroundPage', callback: (params: BrowserContextBackgroundPageEvent) => void): this;
   on(event: 'serviceWorker', callback: (params: BrowserContextServiceWorkerEvent) => void): this;
+  on(event: 'request', callback: (params: PageRequestEvent) => void): this;
```

Comment:
Instead of manually changing this file, update the `protocol.yml` with the new event. If you run `npm run watch` to compile code in the background, it should regenerate this file for you.